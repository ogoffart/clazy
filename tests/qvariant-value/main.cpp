#include <QtCore/QVariant>
#include <QtCore/QUrl>


enum Foo {
    Foo1
};
Q_DECLARE_METATYPE(Foo)


void test()
{
    QVariant v;
    v.toUrl(); //ok
    v.toBool(); //ok
    v.value<QUrl>(); //warn
    v.value<bool>(); //warn
    v.value<float>(); //warn
    v.value<int>(); //warn
    v.value<Foo>(); //ok
    QVariant(45).value<long int>();
}
