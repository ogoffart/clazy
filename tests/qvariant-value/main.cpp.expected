qvariant-value/main.cpp:16:5: warning: Use qvariant_cast instead of QVariant::value [-Wclazy-qvariant-value]
qvariant-value/main.cpp:17:5: warning: Use qvariant_cast instead of QVariant::value [-Wclazy-qvariant-value]
qvariant-value/main.cpp:18:5: warning: Use qvariant_cast instead of QVariant::value [-Wclazy-qvariant-value]
qvariant-value/main.cpp:19:5: warning: Use qvariant_cast instead of QVariant::value [-Wclazy-qvariant-value]
qvariant-value/main.cpp:20:5: warning: Use qvariant_cast instead of QVariant::value [-Wclazy-qvariant-value]
qvariant-value/main.cpp:21:5: warning: Use qvariant_cast instead of QVariant::value [-Wclazy-qvariant-value]
